pennylane/gradients/pulse_generator_gradient.py

```diff
@@ -0,0 +1,229 @@
+        Return the real and imaginary parts separately."""
+        mat = _compute_matrix(*args)
+        return mat.real, mat.imag
+
+    # Compute the Jacobian of _compute_matrix, giving the Jacobian of the real and imag parts
+    # The output is a tuple, with one entry per parameter, each of which has the axes
+    # (mat_dim, mat_dim, *parameter_shape)
+    jac_real, jac_imag = jax.jacobian(_compute_matrix_split, argnums=0)(op.data)
+
+    # Compute the matrix of the pulse itself and conjugate it. Skip the transposition of the adjoint
+    # The output has the shape (mat_dim, mat_dim)
+    U_dagger = qml.math.conj(_compute_matrix([qml.math.detach(d) for d in op.data]))
+
+    # Compute U^\dagger @ \partial U / \partial \theta_k
+    # For each entry ``j`` in the tuple ``jac``,
+    # 1. Contract U_dagger with j along the axes [0], [0]. This effectively transposes
+    #    U_dagger, which we skipped above.
+    # 2. After contraction, the axes are (mat_dim, mat_dim, *parameter_shape).
+    # 3. Move the first two axis to the last two positions.
+    moveax = partial(qml.math.moveaxis, source=(0, 1), destination=(-2, -1))
+    return tuple(
+        moveax(qml.math.tensordot(U_dagger, j_r + 1j * j_i, axes=[[0], [0]]))
+        for j_r, j_i in zip(jac_real, jac_imag)
+    )
+
+
+def _one_parameter_paulirot_coeffs(generators, num_wires):
+    r"""Compute the Pauli coefficients of the one-parameter group generators that reproduce
+    the partial derivatives of a parametrized evolution. The coefficients correspond
+    to the decomposition into (rescaled) Pauli word generators as used by ``PauliRot``
+    gates.
+
+    Args:
+        generators (tuple[tensor_like]): Generators of the one-parameter groups that
+            reproduce the partial derivatives of the parametrized evolution of interest.
+        num_wires (int): Number of wires the parametrized evolution acts on.
+
+    Returns:
+        tuple[tensor_like]: Coefficients of the provided generators in the Pauli basis,
+        modified by a factor of ``2j`` (see warning below).
+
+    .. warning::
+
+        This function includes a prefactor ``2j`` in its output compared to the "standard" Pauli
+        basis coefficients. That is, for the effective generator :math:`\frac{1}{3}iX`, which has
+        the Pauli basis coefficient :math:`\frac{1}{3}i`, this function will compute the
+        number :math:`-\frac{2}{3}`. This is in order to accomodate the use of ``qml.PauliRot``
+        in the pulse generator differentiation pipeline.
+
+    """
+    # The generators are skew-Hermitian. Therefore _pauli_decompose will return a purely
+    # imaginary tensor. Multiplying with 2i results in a real-valued tensor, which
+    # we prefer over a complex-valued tensor with vanishing imaginary part
+    return tuple(qml.math.real(2j * _pauli_decompose(g, num_wires)) for g in generators)
+
+
+def _nonzero_coeffs_and_words(coefficients, num_wires, atol=1e-8):
```

Comment:
Good question! I think I would not go into this here, because PennyLane mostly _allows_ different precisions but does not act on them regarding the precisions used in PL methods. But this might be a nice topic to discuss with core?